app/controllers/users_controller.rb

```diff
@@ -58,14 +58,18 @@ def edit
 
   def update
     @user = @current_user
-
     user_params = permitted_user_params
+
     if user_params[:password].present? && !@user.valid_password?(user_params[:current_password])
       flash[:error] = "Current password is incorrect."
       return render :edit
     end
     user_params.delete :current_password
 
+    if user_params[:professor_verification_requested] == "1" && !@user.professor_verification_requested
```

Comment:
Since this updates the professor user's `professor_verification_requested` attribute, does this mean a professor can't request this twice? Would that cause problems like if the email got lost or the link somehow became invalid?

Would it be better to deny them verification only if they're already verified, as opposed to requesting to be verified?

```ruby
if user_params[:professor_verification_requested] == "1" && !@user.verified_professor?
```

Reply:
For now this feels like the best option to start with. The verification option still valid would be confusing to professors/TAs. Certainly email failing is a concern, but I think we'll go with this and then see if it doesn't work.